src/code42cli/click_ext/types.py

```diff
@@ -16,3 +20,98 @@ def convert(self, value, param, ctx):
             return file.read()
         else:
             return value
+
+
+class MagicDate(click.ParamType):
+    """Declares a parameter to be a 'magic' date string. Accepts an optional `round` argument
+    which can be a function that takes a datetime and returns it rounded appropriately. This allows
+    imprecise "day" input values (2020-01-01, 3d) to be rounded to the start or end of the day
+    if needed. Accepts the following values as user input:
+
+    timestamp formats:
+        yyyy-MM-dd
+        yyyy-MM-dd HH
+        yyyy-MM-dd HH:MM
+        yyyy-MM-dd HH:MM:SS
+
+    short-string (day, hour, min) formats:
+        30d
+        24h
```

Comment:
Do we also want to support 24H? (case-insensitve)

Reply:
yeah, that's simple enough to include

Reply:
I think you need to include it in the regex somehow for it to work